h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -808,15 +804,11 @@ public int compare(Chunk one, Chunk two) {
             validIds.set(c.id);
 
             try {
-                MVMap<String, String> oldMeta = meta.openReadOnly(c.metaRootPos, c.version);
+                Cursor<String, Object> cursor = new Cursor<>(readPage(meta, c.metaRootPos), "chunk.");
                 boolean valid = true;
-                for(Iterator<String> iter = oldMeta.keyIterator("chunk."); valid && iter.hasNext(); ) {
-                    String s = iter.next();
-                    if (!s.startsWith("chunk.")) {
-                        break;
-                    }
-                    s = oldMeta.get(s);
-                    valid = validIds.get(Chunk.fromString(s).id);
+                String key;
+                while (valid && cursor.hasNext() && (key = cursor.next()).startsWith("chunk.")) {
+                    valid = validIds.get((int) Long.parseLong(key.substring("chunk.".length()), 16));
```

Comment:
Description of chuck (associated value) is not checked here any more. I'm not sure that such check was useful, however. Is it enough to check that keys can be read and that they are valid?